### PR TITLE
Dora-LeRobot recording

### PR DESCRIPTION
**Pull request by Hennzau**
# Structure

- [x] `aloha`, `reachy`, `alexk_lcr` and `so100` moved to a `robots` subdirectory
- [x] Improving README.md
- [x] Improving requirements.txt
- [x] `dora_lerobot` local library: temporary local library that contains usefuls objects that will be easily accessible from LeRobot when LeRobot is ready
- [x] Add steps inside `INSTALLATION.md` to install `dora-record`

# Code

- [x] Add everything to teleoperate `so100` with `alexk_lcr` leader arm
- [x] Add a simple mujoco simulation
- [x] Add recording pipeline for `alexk_lcr` (new nodes in `node_hub`)
- [x] Index data from servos with a name : associate a list of joint names with the list of values (position, velocity...)
- [ ] ~Add `gym_lowcostrobot` environment (with help from Julien)~ (next PR)

# Node Hub

- [x] `mujoco_client`: you can pass the PATH to load for simulation, and as other arm clients, you can pass goal_position with joint_names
- [x] `video_encoder`: input an image, and episode status, it will encode all images into a video.
- [x] `lerobot_record`: input ticks, gather images and display them in a simple interface. Manage keyboard inputs for instructions (episode, fail)
- [x] `replay_episode`

# Script

- [x] ~`convert_dora_parquet_to_lerobot`: use the script from remi to convert a recording in lerobot format~ (next PR)
- [x] ~`purge_episode`: remove failed episode, merge data into one parquet file~
- [x] export recording output to an episode (raw episode + FPS episode)
- [x] camera 30Hz fix ?

### CODE REVIEW

File: robots/so100/requirements.txt
Comment (haixuanTao) [inline robots/so100/requirements.txt:1]: Super neat way of managing nested requirements!

File: gym_dora/gym_dora/env_replay.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import pyarrow as pa
 from dora import Node
 from gymnasium import spaces
-from lerobot.common.datasets.lerobot_dataset import LeRobotDataset
+from dora_lerobot.common.datasets.lerobot_dataset import LeRobotDataset
```
Comment (haixuanTao) [inline gym_dora/gym_dora/env_replay.py]: I think this should be lerobot right?

File: robots/reachy/nodes/reachy_vision_client.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 import numpy as np
 import pyarrow as pa
 from dora import Node
-from lerobot.common.datasets.lerobot_dataset import LeRobotDataset
+from dora_lerobot.common.datasets.lerobot_dataset import LeRobotDataset
```
Comment (haixuanTao) [inline robots/reachy/nodes/reachy_vision_client.py]: Same question as above

**Comment by Hennzau** [comment]
I don't think I can do more on this PR, tell me if you need help to review this

EDIT: After chatting with Rémi, I have to change my current script that exports a recording into a dataset. But that's not a hard task

**Review by haixuanTao — APPROVED** [review]
Looks good overall!

File: gym_dora/gym_dora/env_replay.py (hunk 1 above)
Comment (Hennzau) [reply gym_dora/gym_dora/env_replay.py]: Totally right. My IDE automatically replaced it sorry

File: robots/alexk_lcr/INSTALLATION.md [hunk 3]
```diff
@@ -23,17 +23,17 @@ cargo install dora-cli
...
 ```bash
 git clone https://github.com/dora-rs/dora-lerobot
 ```
 
-- Open a bash terminal and navigate to the root of the submodule `alexk_lcr` in the repository.
+- Open a bash terminal and navigate to the repository by running the following command:
 
 ```bash
-cd dora-lerobot/alexk_lcr
+cd dora-dora_lerobot
```
Comment (haixuanTao) [inline robots/alexk_lcr/INSTALLATION.md]:
```suggestion
cd dora-lerobot
```
Maybe?

File: robots/alexk_lcr/RECORDING.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+- Adjust the serial ports of `lcr_leader` and `lcr_follower` in the `record_mono_teleop_real.yml` file.
+- Adjust the camera ID in the `record_mono_teleop_real.yml` file.
+- Adjust camera and video WIDTH and HEIGHT in the `record_mono_teleop_real.yml` file, if needed.
+- Adjust OFFSETS and DRIVE_MODES environment variables in the `record_mono_teleop_real.yml` file for both arms.
+
+You can now start the Dora pipeline to record episodes for LeRobot:
+
+```bash
+cd dora-dora_lerobot
```
Comment (haixuanTao) [inline robots/alexk_lcr/RECORDING.md]: Same as above

**Review by haixuanTao — COMMENTED** [review]
It seems that there might be couple of lerobot -> dora_lerobot renaming within READMEs

File: robots/alexk_lcr/INSTALLATION.md (hunk 3 above)
Comment (Hennzau) [reply robots/alexk_lcr/INSTALLATION.md]: Thank you, I hadn't noticed all these issues that my Refactoring tool caused :'(, I'll go back over it and fix everything.

Comment (haixuanTao) [reply robots/alexk_lcr/INSTALLATION.md]: No props, it happens :)

**Pull request merged**